runtime/acl/acl.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package acl
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	aclapi "github.com/fluxcd/pkg/apis/acl"
+)
+
+// Authorization is an ACL helper for asserting access to cross-namespace references.
+type Authorization struct {
+	Client client.Client
+}
```

Comment:
Wondering if we want to make the client publicly accessible, or instead rely on a factory method and make it private, ensuring we only expose `HasAccessToRef` (and not the full client).